ci/deploy_mz_lsp_server/deploy_util.py

```diff
@@ -0,0 +1,105 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+import os
+import tarfile
+import tempfile
+import time
+from pathlib import Path
+
+try:
+    from semver.version import Version
+except ImportError:
+    from semver import VersionInfo as Version  # type: ignore
+
+import boto3
+import humanize
+
+from materialize import git
+
+APT_BUCKET = "materialize-apt"
+#  "materialize-binaries-staging"
+BINARIES_BUCKET = "materialize-binaries-staging"
+TAG = os.environ["BUILDKITE_TAG"]
+VERSION = Version.parse(TAG.removeprefix("mz-lsp-server-v"))
+
+
+def _tardir(name: str) -> tarfile.TarInfo:
+    d = tarfile.TarInfo(name)
+    d.mtime = int(time.time())
+    d.mode = 0o40755
+    d.type = tarfile.DIRTYPE
+    return d
+
+
+def _sanitize_tarinfo(tarinfo: tarfile.TarInfo) -> tarfile.TarInfo:
+    tarinfo.uid = tarinfo.gid = 0
+    tarinfo.uname = tarinfo.gname = "root"
+    return tarinfo
+
+
+def upload_tarball(tarball: Path, platform: str, version: str) -> None:
+    s3_object = f"mz-lsp-server-{version}-{platform}.tar.gz"
+    boto3.client("s3").upload_file(
+        Filename=str(tarball),
+        Bucket=BINARIES_BUCKET,
+        Key=s3_object,
+    )
+    if "aarch64" in platform:
```

Comment:
Could you add a comment why we need this special handling? We need to upload the file with both names?